src/chainstate/stacks/db/blocks.rs

```diff
@@ -5688,7 +5761,14 @@ impl StacksChainState {
             true,
         )?;
 
-        chainstate_tx.commit().map_err(Error::DBError)?;
+        // this will panic if the Clarity commit fails.
```

Comment:
Can you open an issue to execute the `chainstate_tx.commit()` and `clairty_commit.commit()` calls in the same sqlite transaction?  This is ultimately what we really want -- corruption due to one commit succeeding and another failing, or even stalling for a long time in-between, are both undesirable behaviors.